impl/src/main/java/org/ehcache/config/builders/CacheConfigurationBuilder.java

```diff
@@ -63,27 +61,44 @@
   private Expiry<? super K, ? super V> expiry;
   private ClassLoader classLoader = null;
   private EvictionVeto<? super K, ? super V> evictionVeto;
-  private ResourcePools resourcePools = newResourcePoolsBuilder().heap(Long.MAX_VALUE, EntryUnit.ENTRIES).build();
+  private ResourcePools resourcePools;
   private Class<? super K> keyType;
   private Class<? super V> valueType;
 
   /**
    * Creates a new instance ready to produce a {@link CacheConfiguration} with key type {@code <K>} and with value type
-   * {@code <V>}.
+   * {@code <V>} and which will use the {@link ResourcePools configured resources}.
    *
    * @param keyType the key type
    * @param valueType the value type
+   * @param resourcePools the resources to use
    * @param <K> the key type
    * @param <V> the value type
    * @return a {@code CacheConfigurationBuilder}
    */
-  public static <K, V> CacheConfigurationBuilder<K, V> newCacheConfigurationBuilder(Class<K> keyType, Class<V> valueType) {
-    return new CacheConfigurationBuilder<K, V>(keyType, valueType);
+  public static <K, V> CacheConfigurationBuilder<K, V> newCacheConfigurationBuilder(Class<K> keyType, Class<V> valueType, ResourcePools resourcePools) {
+    return new CacheConfigurationBuilder<K, V>(keyType, valueType, resourcePools);
   }
 
-  private CacheConfigurationBuilder(Class<K> keyType, Class<V> valueType) {
+  /**
+   * Creates a new instance ready to produce a {@link CacheConfiguration} with key type {@code <K>} and with value type
+   * {@code <V>} and which will use the {@link ResourcePools configured resources}, passed as a {@link ResourcePoolsBuilder}.
+   *
+   * @param keyType the key type
+   * @param valueType the value type
+   * @param resourcePoolsBuilder the resources to use, as a builder
+   * @param <K> the key type
+   * @param <V> the value type
+   * @return a {@code CacheConfigurationBuilder}
+   */
+  public static <K, V> CacheConfigurationBuilder<K, V> newCacheConfigurationBuilder(Class<K> keyType, Class<V> valueType, Builder<ResourcePools> resourcePoolsBuilder) {
```

Comment:
If the Builder is typed as Builder<ArbitraryResourcePoolsSubtype> then this will not compile... even though the built result could be passed to the other method.